src/main/java/com/aws/greengrass/clientdevices/auth/certificate/infra/ClientCertificateStore.java

```diff
@@ -5,108 +5,99 @@
-     * @throws KeyStoreException - If fails to store the key store into disk
+     * @throws IOException - If fails to write PEM
      */
-    public void storePem(String certificateId, String certificatePem) throws CertificateException, KeyStoreException {
-        CertificateFactory cf = CertificateFactory.getInstance("X.509");
-        InputStream targetStream = new ByteArrayInputStream(certificatePem.getBytes());
-        Certificate cert = cf.generateCertificate(targetStream);
-        keyStore.setCertificateEntry(certificateId, cert);
+    public void storePem(String certificateId, String certificatePem) throws IOException {
+        saveCertificatePem(certificateIdToPath(certificateId), certificatePem);
     }
 
     /**
      * Removes the PEM for a certificateId alias.
      * @param certificateId - a certificate id
-     * @throws KeyStoreException - if the keystore has not been initialized, or if the entry cannot be removed.
+     * @throws IOException - if the certificate cannot be removed.
      */
-    public void removePem(String certificateId) throws KeyStoreException {
-        keyStore.deleteEntry(certificateId);
+    public void removePem(String certificateId) throws IOException {
+        Path certPath = certificateIdToPath(certificateId);
+        Files.deleteIfExists(certPath);
     }
 
     /**
      * Returns the PEM for a certificate.
      *
      * @param certificateId - The id of a Certificate
+     * @throws IOException - if the certificate exists but cannot be loaded
      */
-    public Optional<String> getPem(String certificateId) {
-        if (certificateId == null) {
+    public Optional<String> getPem(String certificateId) throws IOException {
+        if (!exists(certificateId)) {
             return Optional.empty();
         }
 
-        try {
-            X509Certificate cert = (X509Certificate) keyStore.getCertificate(certificateId);
-
-            if (cert == null) {
-                return Optional.empty();
-            }
+        return Optional.of(loadDeviceCertificate(certificateId));
+    }
 
-            return Optional.of(CertificateHelper.toPem(cert));
-        } catch (KeyStoreException | CertificateEncodingException | IOException e) {
-            return Optional.empty();
+    private void saveCertificatePem(Path filePath, String certificatePem) throws IOException {
+        Utils.createPaths(filePath.getParent());
+        try (OutputStream writeStream = Files.newOutputStream(filePath)) {
+            writeStream.write(certificatePem.getBytes());
         }
     }
+
+    private Path certificateIdToPath(String certificateId) {
+        return workPath.resolve(CLIENTS_DIRECTORY).resolve(certificateId + ".pem");
+    }
+
+    private String loadDeviceCertificate(String certificateId) throws IOException {
+        Path certPath = certificateIdToPath(certificateId);
+        return new String(Files.readAllBytes(certPath));
```

Comment:
should we specify character set here? The default character-set it uses varies across platform (and users).